hydra-tui/test/Hydra/TUISpec.hs

```diff
@@ -40,61 +40,65 @@ import Hydra.ContestationPeriod (toNominalDiffTime)
-          shouldRender "Initializing"
-          sendInputEvent $ EvKey (KChar 'a') []
-          threadDelay 1
-          shouldRender "Idle"
-          sendInputEvent $ EvKey (KChar 'q') []
+        it "supports the init & abort Head life cycle" $
+          \TUITest{sendInputEvent, shouldRender} -> do
+            threadDelay 1
+            shouldRender "connected"
+            shouldRender "Idle"
+            sendInputEvent $ EvKey (KChar 'i') []
+            threadDelay 1
+            shouldRender "Initializing"
+            sendInputEvent $ EvKey (KChar 'a') []
+            threadDelay 1
+            shouldRender "Idle"
+            sendInputEvent $ EvKey (KChar 'q') []
 
-      it "supports the full Head life cycle" $
-        \TUITest{sendInputEvent, shouldRender} -> do
-          threadDelay 1
-          shouldRender "connected"
-          shouldRender "Idle"
-          sendInputEvent $ EvKey (KChar 'i') []
-          threadDelay 1
-          shouldRender "Initializing"
-          sendInputEvent $ EvKey (KChar 'c') []
-          threadDelay 1
-          shouldRender "42000000 lovelace"
-          sendInputEvent $ EvKey (KChar ' ') []
-          sendInputEvent $ EvKey KEnter []
-          threadDelay 1
-          shouldRender "Open"
-          sendInputEvent $ EvKey (KChar 'c') []
-          threadDelay 1
-          shouldRender "Closed"
-          shouldRender "Remaining time to contest"
-          threadDelay (realToFrac $ toNominalDiffTime tuiContestationPeriod + gracePeriod + someTime)
-          sendInputEvent $ EvKey (KChar 'f') []
-          threadDelay 1
-          shouldRender "Final"
-          shouldRender "42000000 lovelace"
-          sendInputEvent $ EvKey (KChar 'q') []
+        it "supports the full Head life cycle" $
+          \TUITest{sendInputEvent, shouldRender} -> do
+            threadDelay 1
+            shouldRender "connected"
+            shouldRender "Idle"
+            sendInputEvent $ EvKey (KChar 'i') []
+            threadDelay 1
+            shouldRender "Initializing"
+            sendInputEvent $ EvKey (KChar 'c') []
+            threadDelay 1
+            shouldRender "42000000 lovelace"
+            sendInputEvent $ EvKey (KChar ' ') []
+            sendInputEvent $ EvKey KEnter []
+            threadDelay 1
+            shouldRender "Open"
+            sendInputEvent $ EvKey (KChar 'c') []
+            threadDelay 1
+            shouldRender "Closed"
+            shouldRender "Remaining time to contest"
+            threadDelay (realToFrac $ toNominalDiffTime tuiContestationPeriod + gracePeriod + someTime)
+            sendInputEvent $ EvKey (KChar 'f') []
+            threadDelay 1
+            shouldRender "Final"
+            shouldRender "42000000 lovelace"
+            sendInputEvent $ EvKey (KChar 'q') []
+    it "should format time with whole values for every unit, not total values" $ do
+      let time = (3675 :: NominalDiffTime)
+      timeFormatted time `shouldBe` "0d 1h 1m 15s"
```

Comment:
As it's basically a one liner to do these tests, we could think about adding some more. What about a time which involves days? What about 0 or negative times?

Reply:
Great catch !!

By testing many days as it was overflowing according to days of the week.